pkg/controller/llmisvc/router.go

```diff
@@ -207,8 +205,68 @@ func RouterLabels(llmSvc *v1alpha1.LLMInferenceService) map[string]string {
 	}
 }
 
-func semanticHTTPRouteIsEqual(e *gatewayapi.HTTPRoute, c *gatewayapi.HTTPRoute) bool {
+func semanticHTTPRouteIsEqual(e *gwapiv1.HTTPRoute, c *gwapiv1.HTTPRoute) bool {
 	return equality.Semantic.DeepDerivative(e.Spec, c.Spec) &&
 		equality.Semantic.DeepDerivative(e.Labels, c.Labels) &&
 		equality.Semantic.DeepDerivative(e.Annotations, c.Annotations)
 }
+
+// EvaluateGatewayConditions evaluates the readiness of all Gateways referenced by the LLMInferenceService
+// and updates the Router condition accordingly
+func (r *LLMInferenceServiceReconciler) EvaluateGatewayConditions(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx).WithName("evaluateGatewayConditions")
+
+	// If no router or gateway configuration, skip Gateway evaluation
+	if llmSvc.Spec.Router == nil || llmSvc.Spec.Router.Gateway == nil || !llmSvc.Spec.Router.Gateway.HasRefs() {
+		logger.Info("No Gateway references found, skipping Gateway condition evaluation")
+		return nil
+	}
+
+	gateways, err := r.FetchReferencedGateways(ctx, llmSvc)
+	if err != nil {
+		llmSvc.MarkRouterNotReady("GatewayFetchError", "Failed to fetch referenced Gateways: %v", err.Error())
+		return fmt.Errorf("failed to fetch referenced gateways: %w", err)
+	}
+
+	notReadyGateways := evaluateGatewayReadiness(ctx, gateways)
+
+	if len(notReadyGateways) > 0 {
+		llmSvc.MarkRouterNotReady("GatewaysNotReady", "One or more referenced Gateways are not ready")
+		return nil
+	}
+	llmSvc.MarkRouterReady()
+	logger.Info("All referenced Gateways are ready")
+
+	return nil
+}
+
+// FetchReferencedGateways retrieves all Gateway objects referenced in the LLMInferenceService spec
+func (r *LLMInferenceServiceReconciler) FetchReferencedGateways(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) ([]*gwapiv1.Gateway, error) {
```

Comment:
for consistency with `collectReferencedRoutes`, I"d call this `collectReferencedGateways`